com.unity.render-pipelines.high-definition/Editor/Material/Nature/HDSpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,65 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    /// <summary>
+    /// SpeedTree8 material upgrader for HDRP.
+    /// </summary>
+	class HDSpeedTree8MaterialUpgrader : SpeedTree8MaterialUpgrader
+    {
+        /// <summary>
+        /// Creates a SpeedTree8 material upgrader for HDRP.
+        /// </summary>
+        /// <param name="sourceShaderName">Original shader name.</param>
+        /// <param name="destShaderName">Upgrade shader name.</param>
+        public HDSpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName)
+			: base(sourceShaderName, destShaderName, HDSpeedTree8MaterialFinalizer)
+		{
+        }
+
+        public static void HDSpeedTree8MaterialFinalizer(Material mat)
+        {
+            SetHDSpeedTree8Defaults(mat);
+            HDShaderUtils.ResetMaterialKeywords(mat);
+        }
+        /// <summary>
+        /// Checks if a given material is an HD SpeedTree8 material.
+        /// </summary>
+        /// <param name="mat">Material to check.</param>
+        /// <returns></returns>
+        public static bool IsHDSpeedTree8Material(Material mat)
+        {
+            return (mat.shader.name == "HDRP/Nature/SpeedTree8");
+        }
+
+        /// <summary>
+        /// Restores SpeedTree8-specific material properties and keywords that were set during import and should not be reset.
+        /// </summary>
+        /// <param name="mat">SpeedTree8 material.</param>
+        public static void RestoreHDSpeedTree8Keywords(Material mat)
+        {
+            int wq = (int)mat.GetFloat("_WINDQUALITY");
+            mat.EnableKeyword(WindQualityString[wq]);
```

Comment:
just for completeness, I don't think this is necessary because i believe it's already done by the applymaterialdrawer (quite sure but that may be worth testing).
It can be interesting to remove because if the billboard code below gets removed in next version, the function could be deleted altogether

Reply:
I confirmed it imports fine without it. :) Looks like streamlining the code with the upgrader solved a lot of my problems.
Removed in 7eab867